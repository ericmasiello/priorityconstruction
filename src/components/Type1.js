import { TYPE_SIZE } from '../styles/vars';
import typeComponentFactory from '../utils/typeComponentFactory';

const result = typeComponentFactory(TYPE_SIZE.t1, {
  defaultTag: 'h1',
  displayName: 'Type1',
});

export const RawType = result.Type;

export default result.StyledType;
