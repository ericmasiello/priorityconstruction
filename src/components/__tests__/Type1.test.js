import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Type1, { RawType } from '../Type1';

it('should render', () => {
  const component = renderer.create(<Type1 />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render children', () => {
  const wrapper = shallow(<RawType>Test</RawType>);

  expect(wrapper.text()).toBe('Test');
});

it('should render additional props', () => {
  const wrapper = shallow(<RawType data-test="my-attribute" />);

  expect(wrapper.find('[data-test="my-attribute"]')).toHaveLength(1);
});

it('should display additional classes', () => {
  const wrapper = shallow(<RawType className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render as a <h1> by default', () => {
  const wrapper = shallow(<RawType />);

  expect(wrapper.type()).toBe('h1');
});

it('should render with a custom tag', () => {
  const wrapper = shallow(<RawType tag="span" />);

  expect(wrapper.type()).toBe('span');
});

