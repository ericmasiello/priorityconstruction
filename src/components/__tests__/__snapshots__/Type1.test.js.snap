// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.c0 {
  font-size: 10vw;
  line-height: 1.2;
}

@media (max-width:10rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:31.57334518321rem) {
  .c0 {
    font-size: 3.157334518321rem;
  }
}

<h1
  className="c0"
/>
`;
